frontend/apps/api_users/logging_helpers.py

```diff
@@ -0,0 +1,99 @@
+import csv
+import logging
+from dataclasses import dataclass
+from io import StringIO
+
+import boto3
+from django.conf import settings
+
+logger = logging.getLogger()
+
+
+@dataclass
+class APIKeyForLogging:
+    key: str
+    key_name: str
+    user_name: str
+    email: str
+    usage_reason: str
+
+    @property
+    def dc_environment(self):
+        return getattr(settings, "DC_ENVIRONMENT", None)
+
+    @property
+    def file_name(self):
+        return f"{self.user_name.replace(' ','_') + '--' if self.user_name  else ''}{self.key_name.replace(' ','_')}--{self.key[:4]}.csv"
```

Comment:
If we're joining on the actual API key, shouldn't that full key be the file name? Imagine we end up renaming a user for some reason, or a user makes more than one key...will this end up causing duplicate rows?

Reply:
Yep, so more than one key would've been ok, but not renaming a user. With the fixup it's one file per key value.

Reply:
[7b57072](https://github.com/DemocracyClub/aggregator-api/pull/607/commits/7b57072a50b627fd357d20d1581ee7a5baf2076d)